packages/loopring_v3/contracts/amm/LoopringAmmPool.sol

```diff
@@ -119,4 +130,29 @@ contract LoopringAmmPool is
     {
         state.withdrawWhenOffline();
     }
+
+    function canDrain(address drainer, address token)
+        public
+        override
+        view
+        returns (bool)
+    {
+        if (drainer != state.exchange.owner()) {
+            return false;
+        }
+
+        // When online, owner can claim all tokens, as we do not allow layer-1 joins.
```

Comment:
We can never allow draining the pool token itself here! Even when online, if people do a forced exit, they transfer their pool tokens, the amount they want to burn, to this contract. So still needs this part here in the code above this point:
```
// Never allow draining the pool token, these can be stored in this contract for forced exits at all times
if (token == address(this)) {
    return false;
}
```